src/event.rs

```diff
@@ -421,3 +445,20 @@ macro_rules! event_subtype {
         }
     }
 }
+
+extern "C" fn event_handler_trampoline(event: *mut ffi::GdkEvent, ptr: glib_ffi::gpointer) {
+    let mut event = unsafe { from_glib_none(event) };
```

Comment:
These functions should be `unsafe extern "C"` IMHO

Reply:
Yes but that marks the whole function unsafe. Everything in there is unsafe, not just the parts you previously put into an `unsafe` block :)